contingency/contingency-api/src/main/java/com/powsybl/contingency/json/IdentifierContingencyListDeserializer.java

```diff
@@ -34,34 +34,23 @@ public IdentifierContingencyList deserialize(JsonParser parser, DeserializationC
         String name = null;
         String version = null;
         List<NetworkElementIdentifier> networkElementIdentifiers = Collections.emptyList();
-
         while (parser.nextToken() != JsonToken.END_OBJECT) {
             switch (parser.getCurrentName()) {
-                case "version":
+                case "version" -> {
                     version = parser.nextTextValue();
-                    deserializationContext.setAttribute("version", version);
-                    break;
-
-                case "identifiableType":
+                    deserializationContext.setAttribute("identifierListVersion", version);
+                }
+                case "identifiableType" -> {
                     JsonUtil.assertLessThanOrEqualToReferenceVersion(CONTEXT_NAME, "identifiableType", version, "1.0");
                     parser.nextToken();
-                    break;
-
-                case "name":
-                    name = parser.nextTextValue();
-                    break;
-
-                case "type":
-                    parser.nextToken();
-                    break;
-
-                case "identifiers":
+                }
+                case "name" -> name = parser.nextTextValue();
+                case "type" -> parser.nextToken();
```

Comment:
It's quite strange to read the type but ignoring it. In most cases (if not all?), we parse the type before calling this deserialize method. We should at least check that the type is the one expected don't you think?